spring-kafka/src/main/java/org/springframework/kafka/listener/ConcurrentMessageListenerContainer.java

```diff
@@ -202,15 +202,24 @@ public boolean isContainerPaused() {
 
 	@Override
 	public boolean isChildRunning() {
-		if (!isRunning()) {
-			return false;
-		}
-		for (MessageListenerContainer container : this.containers) {
-			if (container.isRunning()) {
+		this.lifecycleLock.lock();
+		try {
+			if(isRunning()) {
+				return true;
+			}
+			if(this.containers.isEmpty() && this.stoppedContainers.get() < this.concurrency) {
```

Comment:
I still don't understand this condition: how it is `true` if we don't have containers?

Reply:
containers are cleared once the stop containers is called. But, still the containers will be running in the background. In that scenario, it will be true.

Reply:
OK. I'm not sure what you try to achieve here, but looks like you are mixing concerns in the PR.
Can we concentrate on one problem at a time?
Looks like we have agreed about a `ConcurrentContainerStoppedEvent`, but you do something else in the PR.
It is very hard to review.